kong/db/init.lua

```diff
@@ -739,30 +740,50 @@ do
                               self.strategy, mig.name)
         end
 
+        if mig.name == "000_base" then
+          local is_0_14, err_0_14, is_gt_0_14, missing_0_14_id, missing_0_14_migration =
+            self.connector:is_0_14()
+
+          if err_0_14 then
+            return nil, fmt_err(self, "unable to detect database version (%s) ", err_0_14)
+          end
+
+          if not is_0_14 and not is_gt_0_14 then
+            if missing_0_14_migration then
+              return nil, fmt_err(self, "migration '%s' can only be executed on " ..
+                                  "a %s database that is migrated to version 0.14.x, " ..
+                                  "but database is missing '%s' migration for '%s'",
+                                  mig.name, self.strategy, missing_0_14_migration,
+                                  missing_0_14_id)
```

Comment:
It seems like in this code path, `missing_0_14_id` may sometimes be `nil`. See https://github.com/Kong/kong/pull/3824/files#diff-aa25782656ddfb572924702f3d4e838cR990, in which we complain that an entire subsystem's migrations have not been executed (in prior framework). Note that we return 4 arguments, not 5. I believe using a `res` table to hold that information will prevent the confusing API with the multiple return values.

I am not sure if the error message is properly written either, as it seems to be expecting a migration's name, but in fact, _all_ migrations of this subsystem are missing, afaict. Thus, the proper error message seems to be:
```
... but [database] is missing migrations for '%s' component.
```